bool enable
dims[] positions
float64[] vel_rate
